pkg/kubelet/kubelet_node_status.go

```diff
@@ -113,6 +114,16 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 			Unschedulable: !kl.registerSchedulable,
 		},
 	}
+	if len(kl.kubeletConfiguration.RegisterWithTaints) > 0 {
+		annotations := make(map[string]string)
+		b, err := json.Marshal(kl.kubeletConfiguration.RegisterWithTaints)
+		if err != nil {
+			return nil, err
+		}
+		annotations[api.TaintsAnnotationKey] = string(b)
+		node.ObjectMeta.Annotations = annotations
```

Comment:
Unless I am missing something, it seems to be overwriting the other annotations a node might have?